ipmi/Sensor_Discovery/scripts/ipmi-sensor-discovery.sh

```diff
@@ -0,0 +1,130 @@
+#
+# This program is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public 
+# License as published  by the Free Software Foundation, version
+# 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License  along with this program.
+# If not, see <http://www.gnu.org/licenses/>.
+#
+# Please submit enhancements, bugfixes or comments via:
+# https://github.com/radiorabe/rabe-zabbix
+#
+# Authors:
+#  Christian Affolter <c.affolter@purplehaze.ch>
+#
+# Description:
+# This script is intended to be used for Zabbix low-level discovery of IPMI
+# sensors, as Zabbix currently lacks LLD for IPMI item.
+#
+# It uses FreeIPMI's "ipmi-sensors" command to produce a list of available
+# sensors and transforms the output into the Zabbix LLD JSON format.
+# One is supposed to run this script as an external script on a Zabbix server
+# or proxy within a LLD rule.
+# The script lists all available sensors of a given IPMI host by default.
+# Optionally, the sensor list can be limited to one or multiple (comma separated)
+# senor types (see "ipmi-sensors --list-sensor-types").
+#
+# Refer to the awk command below for a list of LLD macros.
+#
+# Usage:
+# ipmi-sensor-discovery.sh IPMI-HOST IPMI-USER IPMI-PASSWORD [IPMI-SENSOR-TYPES]
+#
+set -o pipefail
+
+export LC_ALL="C"
+
+IPMI_SENSORS_CMD="/usr/sbin/ipmi-sensors"
+AWK_CMD="/usr/bin/awk"
+
+ipmiHost="$1"
+ipmiUser="$2"
+ipmiPassword="$3"
+ipmiSensorTypes="$4"
+
+if test -z "${ipmiHost}"; then
+    echo "Missing IPMI host as the first parameter" >&2
+    exit 1
+fi
+
+if test -z "${ipmiUser}"; then
+    echo "Missing IPMI user as the second parameter" >&2
+    exit 1
+fi
+
+if test -z "${ipmiPassword}"; then
+    echo "Missing IPMI password as the third parameter" >&2
+    exit 1
+fi
+
+# Specifying the IPMI sensor types is optional
+test -z "${ipmiSensorTypes}" && ipmiSensorTypes="all"
+
+for cmd in ${IPMI_SENSORS_CMD} ${AWK_CMD}; do
+    if ! test -x "${cmd}"; then
+        echo "Command ${cmd} not found or executable" >&2 
+        exit 2
+    fi
+done
+
+
+${IPMI_SENSORS_CMD} --driver-type=LAN \
+                    --hostname="${ipmiHost}" \
+                    --username="${ipmiUser}" \
+                    --password="${ipmiPassword}" \
+                    --privilege-level=USER \
+                    --sensor-types="${ipmiSensorTypes}" \
+                    --output-sensor-thresholds \
+                    --ignore-not-available-sensors \
+                    --comma-separated-output \
+                    --no-header-output \
+                    --quiet-cache \
+                    --sdr-cache-recreate 2>/dev/null | /usr/bin/awk '
```

Comment:
Which was definitely not the idea ;-) Fixed and squashed into 205a116ab0f1a44bff2ec6a17926571c0eae7fcf - thanks!